### PR TITLE
sessions: add impersonate_session_id, remove legacy impersonation

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Add impersonate_session_id, remove legacy impersonation.

## Related issues
- https://github.com/pomerium/pomerium-console/issues/1714

## Checklist
- [x] reference any related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Review by desimone — CHANGES_REQUESTED** [review]

File: authenticate/handlers.go [hunk 1]
```diff
@@ -451,12 +451,18 @@ func (a *Authenticate) userInfo(w http.ResponseWriter, r *http.Request) error {
 		s.ID = uuid.New().String()
 	}
 
+	isImpersonated := false
 	pbSession, err := session.Get(ctx, state.dataBrokerClient, s.ID)
```
Comment (desimone) [inline authenticate/handlers.go:455]: This error looks unchecked? What happens if `err!=nil` here?

Comment (desimone) [reply authenticate/handlers.go:455]: e.g. could `pbSession` be `nil` and cause a nil pointer? Probably worth adding a test.

Comment (calebdoxsey) [reply authenticate/handlers.go:455]: no it can't. `Get` methods in protobuf are nil-safe. But sure I can change the code.

Comment (desimone) [reply authenticate/handlers.go:455]: If the answer is no we don't need to update the code.

**Review by desimone — APPROVED** [review]

**Review by wasaga — APPROVED** [review]
👍 

Side note - I don't understand why we do `headers.rego` instead of writing it in Go.

**Comment by calebdoxsey** [comment]
@wasaga we used to do it in go. It was moved to rego with https://github.com/pomerium/internal/issues/278.

**Pull request merged**